### ISSUE TITLE
Alerts not firing even with matching issues

### ISSUE THREAD

**Issue by abdelazaz** [getsentry/self-hosted#2510]
### Environment

self-hosted (https://develop.sentry.dev/self-hosted/)

### Steps to Reproduce

I created a nex internal integration in developper settings for send notification to teams using a behook.
i try to create an alert to send notification. first,  issues can't  be triggered in sentry interface. I have this message "No issues exist for the current query."

conditions:
Set conditions When an event is captured by Sentry and all of the following happens

Action:
Then perform these actions
Send a notification via webhook-sentry

I don't use metrics.

### Expected Result

the issue alerts can be triggered

### Actual Result

No issues exist for the current query."

### Product Area

Alerts

### Version

Sentry 22.8.0

**Comment by azaslavsky** [comment]
Just to clarify the current situation: you create an alert rule with no rules. Then you trigger a new event (not one that you've used previously, but an entirely new one after this new rule-less alert rule was created). You check the logs to ensure that the event showed up on your sentry backend, and see that it showed up in the issues list as a new issue, but your alert is not triggered?

**Comment by abdelazaz** [comment]
Hi azaslavsky

That's exactly it. I create a new event. I can see it in the list of newly created events. But I don't have any triggered alerts. Regarding the alerts, I've tried everything: multiple rules, no rules. But the result is always no triggered alert corresponding to my created event

**Comment by azaslavsky** [comment]
I'm stumped. This works normally in both our SaaS and dogfood instance. Have you tried upgrading to the latest Sentry release?

**Comment by honghainguyen777** [comment]
I am currently facing the same problem for our self-hosted sentry in production too. 
- Alert is functional in our staging environment. 
- Alert was functional in our production 2 weeks ago, but then it just stopped working by its own.

![image](https://github.com/getsentry/self-hosted/assets/59792233/423defc8-a0da-42fd-ad0d-e7d6383a7c03)

![image](https://github.com/getsentry/self-hosted/assets/59792233/8169bff7-1011-46bf-a37c-87c5c6e80042)

**Comment by azaslavsky** [comment]
@honghainguyen777 did you upgrade in that 2 week timeframe?

**Comment by honghainguyen777** [comment]
@azaslavsky no, I didn't, but I did manage to identify its root cause. It appears that my Kafka containers encountered resource limitations, causing them to cease functioning for several hours.This, in turn, resulted in a significant offset discrepancy within the `snuba-post-processor` consumer group. Strangely, this offset lag transformed into a negative value, which ultimately disabled the alert mechanism. To resolve the situation and restore the alert's functionality, I had to reset the offsets to the latest values by using the `kafka-consumer-groups --bootstrap-server 127.0.0.1:9092 --group snuba-post-processor --topic events --reset-offsets --to-latest`. Now it is working well :)
